client/components/command-palette/test/use-command-filter.tsx

```diff
@@ -0,0 +1,53 @@
+import { COMMAND_SEPARATOR, useCommandFilter } from '../use-command-filter';
+
+describe( 'useCommandFilter', () => {
+	const commandFilter = useCommandFilter();
+
+	test( 'should return 1 for the exact match before separator', () => {
+		const search = 'Manage';
+		const beforeSeparator = 'Manage posts';
+		const afterSeparator = 'your blog posts';
+		const value = beforeSeparator + COMMAND_SEPARATOR + afterSeparator;
+		expect( commandFilter( value, search ) ).toBe( 1 );
+	} );
+
+	test( 'should return 1 for the exact match despite the case (lower or upper) before separator', () => {
+		const search = 'MANAGE';
+		const beforeSeparator = 'Manage posts';
+		const afterSeparator = 'your blog posts';
+		const value = beforeSeparator + COMMAND_SEPARATOR + afterSeparator;
+		expect( commandFilter( value, search ) ).toBe( 1 );
+	} );
+
+	test( 'should return 1 for the exact match with multiple words before separator', () => {
+		const search = 'manage posts';
+		const beforeSeparator = 'Manage posts';
+		const afterSeparator = 'your blog posts';
+		const value = beforeSeparator + COMMAND_SEPARATOR + afterSeparator;
+		expect( commandFilter( value, search ) ).toBe( 1 );
+	} );
+
+	test( 'should return 0.5 for the exact match after separator', () => {
+		const search = 'cache';
+		const beforeSeparator = 'Open hosting configuration';
+		const afterSeparator = 'cache hosting manage';
+		const value = beforeSeparator + COMMAND_SEPARATOR + afterSeparator;
+		expect( commandFilter( value, search ) ).toBe( 0.5 );
+	} );
+
+	test( 'should return 0.5 for the exact match after separator despite upper or lower case', () => {
+		const search = 'CACHE';
+		const beforeSeparator = 'Open hosting configuration';
+		const afterSeparator = 'cache hosting manage';
+		const value = beforeSeparator + COMMAND_SEPARATOR + afterSeparator;
+		expect( commandFilter( value, search ) ).toBe( 0.5 );
```

Comment:
@sejas I added test for this and the function for the `filter` does return `0.5` as expected so technically it does what the prop describes (return the number between `0` and `1`) but it does not change the ranking. I think this indicates that there must be an issue with `cmdk`. What are your thoughts?

Reply:
Just to add, I think we should also some tests to check the rendering of the commands when you are rending an actual array since these are fairly simple and only check if the matching works as expected

Reply:
> Just to add, I think we should also some tests to check the rendering of the commands when you are rending an actual array since these are fairly simple and only check if the matching works as expected

Yes! That would be fantastic. If it's not too complicated, we could create a fake array of commands, and then search for something and check if it's in the array or/and if it's a first result.

Reply:
Sounds good, I will report the issue to them and work on the tests today! Thanks for the review!

Reply:
@sejas I search the `cmdk` repo and found the issue with filtering was reported here: https://github.com/pacocoursey/cmdk/issues/118

There is current a PR in progress to fix the issue: https://github.com/pacocoursey/cmdk/pull/182